src/os/bluestore/KernelDevice.cc

```diff
@@ -130,7 +130,6 @@ int KernelDevice::open(const string& p)
   } else {
     size = st.st_size;
   }
-  size &= ~(block_size);
```

Comment:
yes, it should align to block_size.  It should be `size &= ~(block_size - 1)` .
L157 do this correctly, so this line is redundant.